test/specs/templates.e2e.ts

```diff
@@ -27,6 +27,220 @@ describe('Templates', async () => {
+    const treeViewSection = await sidebar.getContent().getSection(projectName);
+    await treeViewSection.expand();
+
+    // Get the matching (visible) items within the tree which contains "apexClass1".
+    const filteredTreeViewItems = await utilities.getFilteredVisibleTreeViewItemLabels(workbench, projectName, 'ApexClass1');
+
+    expect(filteredTreeViewItems.includes('ApexClass1.cls')).toBe(true);
+    expect(filteredTreeViewItems.includes('ApexClass1.cls-meta.xml')).toBe(true);
+  });
+
+  step('Verify the contents of the Apex Class', async () => {
+    const expectedText = [
+      'public with sharing class ApexClass1 {',
+      '    public ApexClass1() {',
+      '',
+      '    }',
+      '}'
+    ].join('\n');
+    const workbench = await browser.getWorkbench();
+    const editorView = await workbench.getEditorView();
+    const textEditor = await editorView.openEditor('ApexClass1.cls') as TextEditor;
+    const textGeneratedFromTemplate = (await textEditor.getText()).trimEnd();
+    expect(textGeneratedFromTemplate).toEqual(expectedText);
+  });
+
+  step('Push the Apex Class', async () => {
+    // Run "SFDX: Push Source to Default Org".
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Org', 5);
+
+    const successNotificationWasFound = await utilities.attemptToFindNotification(workbench, 'SFDX: Push Source to Default Org successfully ran', 10);
+    expect(successNotificationWasFound).toBe(true);
+
+    // Check the output.
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Salesforce CLI', '=== Pushed Source', 10);
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Created  ApexClass1  ApexClass');
+  });
+
+  // Apex Trigger
+  step('Create an Apex Trigger', async () => {
+    // Clear the output panel, then use the Command palette to run, "SFDX: Create Apex Trigger".
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'View: Clear Output', 1);
+    const inputBox = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Create Apex Trigger', 1);
+
+    // Set the name of the new component to ApexTrigger1.
+    await inputBox.setText('ApexTrigger1');
+    await inputBox.confirm();
+    await utilities.pause(1);
+
+    // Select the default directory (press Enter/Return).
+    await inputBox.confirm();
+
+    const successNotificationWasFound = await utilities.attemptToFindNotification(workbench, 'SFDX: Create Apex Trigger successfully ran', 10);
+    expect(successNotificationWasFound).toBe(true);
+
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Salesforce CLI', 'Finished SFDX: Create Apex Trigger', 10);
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('create force-app/main/default/triggers/ApexTrigger1.trigger');
```

Comment:
when checking for paths we need to use `path.join()` so it works on windows too

Reply:
done